File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;
vector<bool>isPrime(1000001,true);

void eratos(int& n){
    isPrime[1]=false;
    for(int i =2 ;i*i<=n;i++){
        if(isPrime[i]){
            for(int j = i*i;j<=n;j+=i){
                isPrime[j]=false;
            }
        }
    }
}
int solution(int n) {
    int answer = 0;
    eratos(n);
    for(int i =1 ; i<=n;i++){
        if(isPrime[i])
            answer++;
    }
    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
from math import sqrt
def isPrime(num):
    if num<=1:
        return False
    for i in range(2,int(sqrt(num))+1):
        if num%i==0:
            return False
    return True

def solution(numbers):
    answer = 0
    n=len(numbers)
    numbers=list(numbers)
    se=set()
    for i in range(1,n+1):
        for order in permutations(range(n),i):
            num=int(''.join(numbers[a] for a in order))
            
            if num in se:
                continue
            if isPrime(num):
                answer+=1
                se.add(num)
    return answer